### DISCUSSION TITLE
V2 Route Stop Patterns

### DISCUSSION THREAD

**Discussion by smohiudd** [transitland/transitland#335] [Q&A] [answered]
I noticed that the V2 Rest API doesn't have the same endpoints as V1. Specifically V2 is lacking route stop patterns and stop distances (something that I found very useful for a project I'm working on). Is there a plan to include stop patterns and/or stop distances in V2 Rest or GraphQL? 

Is there an efficient way to query all stops for a a route ID with the current REST api? I've looked through the documentation and done some testing and it seems I can only query metadata on specific stop ids and nothing associated with route stop patterns.

**Comment by drewda** [comment, marked as answer]
Transitland v2 does internally calculate a replacement for the RouteStopPattern. We have not yet exposed it publicly through the REST API, but likely will do so in the future.

We will likely expand the `/api/v2/rest/stops` endpoint to be able to search by operator or route (like the v1 Datastore API's `served_by` query parameter).

**Reply by drewda** [reply]
We have now added a new query param to the v2 REST stops endpoint to allow queries like:

```
https://transit.land/api/v2/rest/stops?served_by_onestop_ids=o-9q9-bart&apikey=...
https://transit.land/api/v2/rest/stops?served_by_onestop_ids=r-9q9qd2-1&apikey=...
```

For more details, see https://www.transit.land/documentation/rest-api/stops#request-parameters

**Reply by smohiudd** [reply]
Thanks for the update!

**Comment by smohiudd** [comment]
@drewda do you have an update on when you'll be releasing the endpoint for RouteStopPaterns?

**Reply by drewda** [reply]
Hi @smohiudd no update on this, but we'll be glad to post here when we do have an update.
